src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -426,8 +445,35 @@ public Command getDriveToPoseCommand() {
     return AutoBuilder.pathfindToPose(targetPose, constraints, 0.0);
   }
 
+  public void rotationallyLockedIn() {
+    double omega = 0.0;
+
+    // Get current position
+    Translation2d currentPosition = getPose().getTranslation();
+
+    // Calculate desired angle to face the target position
+    double targetAngle =
+        Math.atan2(
+            lockedTargetPosition.getY() - currentPosition.getY(),
+            lockedTargetPosition.getX() - currentPosition.getX());
+
+    // Use PID to rotate toward the target angle
+    omega = angleController.calculate(getRotation().getRadians(), targetAngle);
+    setGoalSpeeds(
+        new ChassisSpeeds(goalSpeeds.vxMetersPerSecond, goalSpeeds.vyMetersPerSecond, omega),
+        false);
+
+    if (DriverStation.getAlliance().isPresent()
+        && DriverStation.getAlliance().get() == Alliance.Red) lockRotationToPosition(redReefCenter);
+    else lockRotationToPosition(blueReefCenter);
```

Comment:
please use curly braces even though you don't technically have to

also this should be in the `alignToFieldElement` function (renamed version of `lockRotationToPosition`) rather than calling it. This does not need to be called every cycle.